extensions/pooling/src/__tests__/acceptance/pooling.acceptance.ts

```diff
@@ -0,0 +1,370 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/pooling
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  BindingKey,
+  Constructor,
+  Context,
+  ContextTags,
+  createBindingFromClass,
+  inject,
+  Provider,
+} from '@loopback/core';
+import {expect} from '@loopback/testlab';
+import {once} from 'events';
+import {Options} from 'generic-pool';
+import {
+  createPooledBindingFactory,
+  PoolingService,
+  PoolingServiceOptions,
+} from '../../';
+import {
+  getPooledValue,
+  Poolable,
+  PooledValue,
+  PoolFactory,
+} from '../../pooling';
+
+describe('Resource pool', () => {
+  const POOLING_SERVICE = BindingKey.create<PoolingService<ExpensiveResource>>(
+    'services.pooling',
+  );
+  let ctx: Context;
+
+  beforeEach(givenContext);
+
+  it('creates a resource pool', async () => {
+    const poolService = await givenPoolService();
+    expect(poolService.pool.size).to.eql(0);
+  });
+
+  it('creates a resource pool as singleton', async () => {
+    const poolService = await givenPoolService();
+    const result = await ctx.get(POOLING_SERVICE);
+    expect(result).to.be.exactly(poolService);
+  });
+
+  it('acquires/releases a resource from the pool', async () => {
+    const poolService = await givenPoolService({max: 5});
+    poolService.start();
```

Comment:
`poolService.start();` is not called in some places. How does it work? This behavior should be documented.

Reply:
My assumption is that `poolService` will not be usable for anything (eg: `poolService.run`) unless it is `start`ed.

Reply:
Quoted from [generic-pool](https://github.com/coopernurse/node-pool#createpool):

> autostart: boolean, should the pool start creating resources, initialize the evictor, etc once the constructor is called. If false, the pool can be started by calling pool.start, otherwise the first call to acquire will start the pool. (default true)